From ef25bf644b7de83849a2f804c84bb54cd2f1255f Mon Sep 17 00:00:00 2001
From: Alison Wang <b18965@freescale.com>
Date: Mon, 11 Apr 2016 17:25:40 +0800
Subject: [PATCH 09/70] armv8: aarch32: Allow RAM to be mapped for LayerScape
 SoC

This patch is based on Kernel v4.1.8. As in v4.1.8, memremap() is not
introduced and the WARN() check is not relaxed to allow MT_MEMORY_RW
mappings of pfn_valid() pages, this patch is needed as a workaround for
spin-table address which locates in RAM to be mapped.

For the latest kernel in upstream, this patch is not needed anymore.

Signed-off-by: Alison Wang <alison.wang@nxp.com>
---
 arch/arm/mm/ioremap.c |    2 ++
 1 file changed, 2 insertions(+)

--- a/arch/arm/mm/ioremap.c
+++ b/arch/arm/mm/ioremap.c
@@ -298,8 +298,10 @@ static void __iomem * __arm_ioremap_pfn_
 	/*
 	 * Don't allow RAM to be mapped - this causes problems with ARMv6+
 	 */
+#ifndef CONFIG_ARCH_LAYERSCAPE
 	if (WARN_ON(pfn_valid(pfn)))
 		return NULL;
+#endif
 
 	area = get_vm_area_caller(size, VM_IOREMAP, caller);
  	if (!area)
